### PR TITLE
Fix layer-browser hot reload

### PR DESCRIPTION
**Pull request by Pessimistress**
Retains app state after reload

### CODE REVIEW

File: examples/layer-browser/src/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      </div>
+    );
+  }
+}
+
+// ---- Main ---- //
+
+const container = document.createElement('div');
+document.body.appendChild(container);
+ReactDOM.render(<Root AppComponent={App} />, container);
```
Comment (balthazar) [inline examples/layer-browser/src/index.js]: could you create an util function like

```js
const render = (AppComponent) =>
  ReactDOM.render(<Root AppComponent={AppComponent} />, container);

render(App);

if (module.hot) {
  module.hot.accept('./app', () => {
    console.log('Hot reloading App component'); // eslint-disable-line
    render(require('./app').default);
  });
}

File: examples/layer-browser/webpack.config.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ const webpack = require('webpack');
 const CONFIG = {
   // bundle app.js and everything it imports, recursively.
   entry: {
-    app: resolve('./src/app.js')
+    app: resolve('./src/index.js')
```
Comment (balthazar) [inline examples/layer-browser/webpack.config.js]: You can `resolve('./src')`

File: examples/layer-browser/src/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/* global document */
+import React, {Component} from 'react';
+import ReactDOM from 'react-dom';
+import App from './app';
+import FPSStats from 'react-stats-zavatta';
+
+class Root extends Component {
+  constructor(props) {
+    super(props);
+    this._appState = null;
```
Comment (balthazar) [inline examples/layer-browser/src/index.js]: I think you can initialize `_appState` as a static property (if the babel config there is permits), not sure how much we want to use these kind of features though

File: examples/layer-browser/src/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/* global document */
+import React, {Component} from 'react';
+import ReactDOM from 'react-dom';
+import App from './app';
+import FPSStats from 'react-stats-zavatta';
```
Comment (balthazar) [inline examples/layer-browser/src/index.js]: I generally put the "vendor" dependencies before the app ones, maybe we could discuss on what we want to do for ordering?

File: examples/layer-browser/src/index.js (hunk 3 above)
Comment (Pessimistress) [reply examples/layer-browser/src/index.js]: It does not

**Pull request merged**